apps/beeai-server/src/beeai_server/auth/utils.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2025 © BeeAI a Series of LF Projects, LLC
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+import logging
+import os
+
+import jwt
+import requests
+from fastapi.security import HTTPAuthorizationCredentials
+from jwt import ExpiredSignatureError, InvalidTokenError, PyJWK
+
+from beeai_server.api.dependencies import ConfigurationDependency
+from beeai_server.configuration import get_configuration
+
+logger = logging.getLogger(__name__)
+
+configuration: ConfigurationDependency = get_configuration()
+
+
+# Initialize JWKS only if OIDC is enabled
+if not configuration.oidc.disable_oidc:
+    audience = configuration.oidc.client_id
+    JWKS_URL = configuration.oidc.jwks_url
+    JWKS_FOLDER = "/tmp/jwks"
+    JWKS_FILE = os.path.join(JWKS_FOLDER, "pubkeys.json")
+
+    os.makedirs(JWKS_FOLDER, exist_ok=True)
+
+    try:
+        response = requests.get(JWKS_URL)
```

Comment:
please don't do network requests on import - we have `bootstrap.py` for initialization